CyberDisc-RichPresence/CDRichPresence.cs

```diff
@@ -1,228 +1,247 @@
+using System.Windows.Forms;
+using System.Windows.Automation;
+using DiscordRPC;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace CyberDisc_RichPresence
+{
+
+    public class CDRichPresence : Form
+    {
+        private NotifyIcon trayIcon;
+        private IContainer components;
+        public DiscordRpcClient client;
+
+        public CDRichPresence()
+        {
+            InitializeComponent();
+            Load += OnLoad;
+            ContextMenu menu = new ContextMenu();
+            menu.MenuItems.Add(0, new MenuItem("Exit", new EventHandler(Exit_click)));
+            trayIcon.ContextMenu = menu;
+        }
+
+        private void OnLoad(object sender, EventArgs e)
+        {
+            Run();
+        }
+
+        private async Task Run()
+        {
+            await Task.Delay(10);
+            Hide();
+            StartRichPresence();
+            trayIcon.ShowBalloonTip(1000);
+        }
+
+        private void OnResize(object sender, EventArgs e)
+        {
+            if (WindowState == FormWindowState.Minimized)
+            {
+                Hide();
+            }
+        }
+
+        private void Exit_click(object sender, EventArgs e)
+        {
+            trayIcon.Dispose();
+            Environment.Exit(0);
+        }
+
+        public void StartRichPresence()
+        {
+            client = new DiscordRpcClient("537710248664760322");
+            client.Initialize();
+            client.SetPresence(new RichPresence()
+            {
+                Details = "Idle",
+                //State = "Idle",
+                Assets = new Assets()
+                {
+                    LargeImageKey = "cd-logo",
+                    LargeImageText = "Cyber Discovery",
+                    SmallImageKey = "cd-logo"
+                }
+            });
+            var timer = new System.Timers.Timer(150);
+            timer.Elapsed += (sender, args) => { UpdateFromURL(); };
+            timer.Start();
+        }
+
+        private void UpdateFromURL()
+        {
+            client.Invoke();
+            string url = CurrentURL();
+            if (url == null || url == "")
+            {
+                client.ClearPresence();
+                return;
+            }
+            if (url.Contains("game"))
+            {
+                DoGameUpdate(url);
+                return;
+            }
+            if (url.Contains("assess"))
+            {
+                DoAssesUpdate(url);
+                return;
+            }
+            if (url.Contains("essentials"))
+            {
+                DoEssentialsUpdates(url);
+                return;
+            }
+            if (url.Contains("elite"))
+            {
+                DoEliteUpdates(url);
+                return;
+            }
+            SetRP("Idle", "", "cd");
+        }
+
+        private void DoEliteUpdates(string url)
+        {
+            SetRP("Working on elite", "Wait what this isnt possible", "game");

```

Comment:
Remove to merge. It's C# so it's already bloated. We don't need more of that.

Reply:
its already in the deployed version, and ima have to say one extra method isnt going to do shit

Reply:
Fun fact: Not your choice

Reply:
i mean its my project....